pennylane/ops/op_math/sum.py

```diff
@@ -0,0 +1,156 @@
+        for s in summands:
+            combined_params += s.parameters
+
+        super().__init__(*combined_params, wires=combined_wires, do_queue=do_queue, id=id)
+        self._name = "Sum"
+
+    def __repr__(self):
+        """Constructor-call-like representation."""
+        return " + ".join([f"{f}" for f in self.summands])
+
+    def __copy__(self):
+        cls = self.__class__
+        copied_op = cls.__new__(cls)
+        copied_op.data = self.data.copy()  # copies the combined parameters
+        copied_op.summands = tuple(s.__copy__() for s in self.summands)
+
+        for attr, value in vars(self).items():
+            if attr not in {"data", "summands"}:
+                setattr(copied_op, attr, value)
+
+        return copied_op
+
+    @property
+    def num_wires(self):
+        return len(self.wires)
+
+    @property
+    def is_hermitian(self):
+        """If all of the terms in the sum are hermitian, then the Sum is hermitian."""
+        return all(s.is_hermitian for s in self.summands)
+
+    def terms(self):
+        return [1.0] * len(self.summands), self.summands
+
+    @property
+    def eigendecomposition(self):
+        """Return the eigendecomposition of the matrix specified by the Hermitian observable.
+
+        This method uses pre-stored eigenvalues for standard observables where
+        possible and stores the corresponding eigenvectors from the eigendecomposition.
+
+        It transforms the input operator according to the wires specified.
+
+        Returns:
+            dict[str, array]: dictionary containing the eigenvalues and the eigenvectors of the operator
+        """
+        Hmat = self.matrix()
+        Hmat = qml.math.to_numpy(Hmat)
+        Hkey = tuple(Hmat.flatten().tolist())
+        if Hkey not in self._eigs:
+            w, U = np.linalg.eigh(Hmat)
+            self._eigs[Hkey] = {"eigvec": U, "eigval": w}
+
+        return self._eigs[Hkey]
+
+    def diagonalizing_gates(self):
+        """Compute diagonalizing_gates (only if op is hermitian)"""
+
+        eigen_vectors = self.eigendecomposition["eigvec"]
+        return [qml.QubitUnitary(eigen_vectors.conj().T, wires=self.wires)]
+
+    def eigvals(self):
+        """Return the eigenvalues of the specified Hermitian observable.
+
+        This method uses pre-stored eigenvalues for standard observables where
+        possible and stores the corresponding eigenvectors from the eigendecomposition.
+
+        Returns:
+            array: array containing the eigenvalues of the Hermitian observable
+        """
+        return self.eigendecomposition["eigval"]
+
+    def matrix(self, wire_order=None):
+        """Representation of the operator as a matrix in the computational basis."""
+
+        def matrix_gen(summands, wire_order=None):
+            """Helper function to construct a generator of matrices"""
+            for op in summands:
+                yield expand_matrix(op.matrix(), op.wires, wire_order=wire_order)
```

Comment:
```suggestion
                yield op.matrix(wire_order=wire_order)
```

`op.matrix` should be able to accept a `wire_order` keyword and do this expansion for you.

Reply:
Hmm, I agree, but I feel like there will be a few operations that don't use the `expand_matrix` logic which will have to be changed. But this is the cleaner way to do it.